### DISCUSSION TITLE
如何实现双语同时显示并候选

### DISCUSSION THREAD

**Discussion by Latomoyocarl** [rime/home#1316] [Q&A]
因为个人经常使用日语，交替打中文和日文时会很麻烦，所以想询问一下有什么方法实现双语同时候选？因为实现不了就画了一个示例图来描述我想要的效果：打字之后同时出现双语的候选，让它默认选择为上方的语言候选，这时按方空格键“[]”可以上下选择想要输入的语言，其余基础操作不改变。这样的话即使是忘记切换输入法了，也不必重新打一遍原话。

（示例图）
![image](https://github.com/rime/home/assets/58652186/5c3c7602-bab5-4421-a49e-4cd3c5511202)

**Comment by pdog18** [comment]
以我目前的经验来看, 无法通过 yaml 配置实现, 只能自己参与修改源码

**Reply by Latomoyocarl** [reply]
可惜我不会修改源代码，所以只好等等看下一个版本会不会实现这个功能了……。

**Comment by fxliang** [comment]
直接使用一个指定的引导按键反查不就可以了吗，毕竟输入之前已经知道是哪个（语言/方案的内容）

**Comment by Lantaio** [comment]
感觉可以借鉴中英混输的套路。

**Comment by MaxHasBeenUsed** [comment]
虽然忘记切输入法也能打出来，但是无论打中文还是日文都会有一行多余不觉得很奇怪吗……

有几个思路希望能对你有用：
1. 中日混输。如果你中文是全拼，日文是罗马音的话，中文和日文的击键特征明显且独特，非常适合做混输。其他情况比如日文是假名输入，那更应该做混输了；中文是双拼或形码的话需要看具体方案，不过应该都只会有少数几个重码。
2. 快捷键切换输入法。看这字体应该是 Windows 系统，在输入法设置里面可以设置快捷键切换***输入法***。我的做法是：Ctrl+Alt+1 是中文输入法、C+A+2 是英文、C+A+3 是日文，然后用软件将 Shift+q 映射到 C+A+1，依次类推。实现的效果就是 S+q/w/e 分别对应中英日三语输入，每次输入前按一下就能保证切到想要的输入法，而不是传统的「在几个输入法间循环」。缺点可能是每个语言都需要单独的输入法。
